shell/platform/windows/flutter_window_unittests.cc

```diff
@@ -229,6 +230,8 @@ TEST(FlutterWindowTest, OnPointerStarSendsDeviceType) {
                           kDefaultPointerDeviceId, WM_LBUTTONDOWN);
   win32window.OnPointerLeave(10.0, 10.0, kFlutterPointerDeviceKindStylus,
                              kDefaultPointerDeviceId);
+
+  win32window.SetView(nullptr);
```

Comment:
Added because `FlutterWindow` will try to send a `HIDE` update to the binding delegate upon its deconstruction, and this test sets the binding delegate to the address of a local stack variable, which means it is no longer valid once execution leaves the scope of this test function. Alternatively, we could replace `delegate` with a `unique_ptr` and get its raw pointer before moving it to pass to `EXPECT_CALL`

Reply:
From a lifetime perspective, if the delegate is always strictly owned by the window and its lifetime is constrained to (at most) the lifetime of the owning window, then using a unique_ptr seems reasonable. We should do a quick audit of the code to be sure. If not, then I'd add a one-line comment explaining the rationale here as a breadcrumb for future archaeologists.

Reply:
Actually, it looks to be the other way around. The delegate i.e. `FlutterWindowsView` owns a pointer to the handler i.e. `FlutterWindow`, but where we're using a `MockWindowBindingHandlerDelegate` in these unit tests, the window is just scoped to the function it's created in. For now I will put a comment here, though moving ownership of `win32window` in these tests to inside the mock delegate may be feasible.